Model/AdyenInitiateTerminalApi.php

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Trigger sync call on terminal
+     * @return mixed
+     * @throws \Exception
+     */
+    public function initiate()
+    {
+        $quote = $this->_checkoutSession->getQuote();
+        $payment = $quote->getPayment();
+        $payment->setMethod(AdyenPosCloudConfigProvider::CODE);
+        $reference = $quote->reserveOrderId()->getReservedOrderId();
+
+        $service = new \Adyen\Service\PosPayment($this->_client);
+        $transactionType = \Adyen\TransactionType::NORMAL;
+        $poiId = $this->_adyenHelper->getPoiId($this->storeId);
+        $serviceID = date("dHis");
+        $initiateDate = date("U");
+        $timeStamper = date("Y-m-d") . "T" . date("H:i:s+00:00");
+        $customerId = $quote->getCustomerId();
+
+        $request = [
+            'SaleToPOIRequest' =>
+                [
+                    'MessageHeader' =>
+                        [
+                            'MessageType' => 'Request',
+                            'MessageClass' => 'Service',
+                            'MessageCategory' => 'Payment',
+                            'SaleID' => 'Magento2Cloud',
+                            'POIID' => $poiId,
+                            'ProtocolVersion' => '3.0',
+                            'ServiceID' => $serviceID,
+                        ],
+                    'PaymentRequest' =>
+                        [
+                            'SaleData' =>
+                                [
+                                    'TokenRequestedType' => 'Customer',
+                                    'SaleTransactionID' =>
+                                        [
+                                            'TransactionID' => $reference,
+                                            'TimeStamp' => $timeStamper,
+                                        ],
+                                ],
+                            'PaymentTransaction' =>
+                                [
+                                    'AmountsReq' =>
+                                        [
+                                            'Currency' => $quote->getCurrency()->getQuoteCurrencyCode(),
+                                            'RequestedAmount' => doubleval($quote->getGrandTotal()),
+                                        ],
+                                ],
+                            'PaymentData' =>
+                                [
+                                    'PaymentType' => $transactionType,
+                                ],
+                        ],
+                ],
+        ];
+
+        // If customer exists add it into the request to store request
+        if (!empty($customerId)) {
+            $shopperEmail = $quote->getCustomerEmail();
+            $recurringContract = $this->_adyenHelper->getAdyenPosCloudConfigData('recurring_type', $this->storeId);
+
+            if (!empty($recurringContract) && !empty($shopperEmail) && !empty($customerId)) {
```

Comment:
CustomerId is already checked if not empty

```suggestion
            if (!empty($recurringContract) && !empty($shopperEmail)) {
```